vignettes/adtr.Rmd

```diff
@@ -0,0 +1,599 @@
+)
+```
+
+## Derive Parameters (TSD) and Analysis Flag `ANL01FL` {#tsd}
+
+Target Lesions Sum of Diameters is the sum of the diameters for lymph and
+non-lymph nodes (sum of analysis values `AVAL` from source observations TLDn &
+TLPn).  The analysis flag `ANL01FL` flags where the number of lesions assessed
+at baseline and at post-baseline match.  To assess whether the number of lesions
+expected and assessed match, you could compare lesion identifiers (`TR.TRLNKID`)
+/use the previous variables set up earlier to compare lesion expected vs lesion
+accessed  or check for target lesion response (from `RS`) =NE.
+
+```{r}
+adtr_sum <- get_summary_records(
+  adtr,
+  by_vars = vars(STUDYID, USUBJID, !!!adsl_vars, AVISIT, AVISITN),
+  analysis_var = AVAL,
+  summary_fun = function(x) sum(x, na.rm = TRUE),
+  set_values_to = vars(
+    PARAMCD = "TSD",
+    PARAM = "Target Lesions Sum of Diameters by Investigator",
+    PARCAT1 = "Target Lesion(s)",
+    PARCAT2 = "Investigator",
+    PARCAT3 = "Recist 1.1"
+  )
+)
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  arrange(adtr_sum, USUBJID, AVISITN),
+  display_vars = vars(USUBJID, PARAMCD, PARAM, AVISIT, AVAL)
+)
+```  
+
+In the below example, the analysis flag, ANL01FL= Y identifies the largest
+target lesion sum of diameter on a particular date for reporting if there were
+multiple observations on the same date and where lesion assessed at
+post-baseline match baseline.
+
+```{r}
+adtr_sum <- adtr_sum %>%
+  derive_var_merged_summary(
+    dataset_add = adtr,
+    by_vars = vars(USUBJID),
+    filter_add = AVISIT == "BASELINE" & str_detect(PARAMCD, "DIAM[12345]"),
+    new_var = LSEXP,
+    analysis_var = TRLNKID,
+    summary_fun = function(x) paste(sort(x), collapse = ", ")
+  ) %>%
+  derive_var_merged_summary(
+    dataset_add = adtr,
+    by_vars = vars(USUBJID, AVISIT),
+    filter_add = str_detect(PARAMCD, "DIAM[12345]") & ANL01FL == "Y",
+    new_var = LSASS,
+    analysis_var = TRLNKID,
+    summary_fun = function(x) paste(sort(x), collapse = ", ")
+  ) %>%
+  derive_var_merged_summary(
+    dataset_add = adtr,
+    by_vars = vars(USUBJID, AVISIT),
+    filter_add = str_detect(PARAMCD, "DIAM[12345]"),
+    new_var = ADY,
+    analysis_var = ADY,
+    summary_fun = function(x) min(x, na.rm = TRUE)
+  ) %>%
+  derive_var_merged_summary(
+    dataset_add = adtr,
+    by_vars = vars(USUBJID, AVISIT),
+    filter_add = str_detect(PARAMCD, "DIAM[12345]"),
+    new_var = ADT,
+    analysis_var = ADT,
+    summary_fun = function(x) min(x, na.rm = TRUE)
+  ) %>%
+  mutate(
+    ANL01FL = if_else(LSEXP == LSASS, "Y", NA_character_)
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  arrange(adtr_sum, USUBJID, AVISITN),
+  display_vars = vars(USUBJID, PARAMCD, AVISIT, LSEXP, LSASS, ADT, ADY)
+)
+```  
+
+## Derive Baseline (`BASETYPE`, `ABLFL`, `BASE`) {#base}
+
+These functions are available in `admiral`, below examples show its usage for `ADTR` 
+
+```{r}
+adtr_sum <- adtr_sum %>%
```

Comment:
Can we not overwrite each object with the same name (e..g `adtr_sum <- adtr_sum`)? or is this a convention?

I find it hard to track and debug with this,  I guess more for code maintenance than usability.

Reply:
@bundfussr

Reply:
For the admiral vignettes we have the convention of using the same dataset name in each step. Thus I would not change it. The convention has the advantage that you can remove or add steps without the need to update the previous and following steps. The downside it that debugging is a bit harder.